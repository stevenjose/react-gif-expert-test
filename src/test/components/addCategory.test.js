import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("probar el componente addCategory", () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategory} />);
  const input = wrapper.find("input");
  
  beforeEach(()=>{
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory setCategories={setCategory} />);
  })
  
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("probar el input ", () => {
    const value = "Simulando value text";
    input.simulate("change", { target: { value } });
  });

  test('No debe postear la informacion con submit', () => {
    input.simulate("change",{ target: { value:'' }});  
    wrapper.find('form').simulate('submit',{preventDefault(){}});
    expect( setCategory ).not.toHaveBeenCalled();
  });

  test('debe llamar a setCategory',()=>{
    const value = "Simulando value text";
    
    wrapper.find('input').simulate('change',{ target:{value} });
    
    wrapper.find('form').simulate('submit',{preventDefault(){}});
    
    expect( setCategory ).toHaveBeenCalled();
    expect( setCategory ).toHaveBeenCalledWith( expect.any(Function));
    expect (input.prop('value').trim()).toBe('');
  })
});
