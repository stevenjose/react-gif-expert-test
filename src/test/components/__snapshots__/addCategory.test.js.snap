// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar el componente addCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="buscar"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
